src/remote.ml

```diff
@@ -361,6 +388,15 @@ let dumpUrgent conn l =
        fillBuffer conn.outputBuffer l >>= fun () ->
        flushBuffer conn.outputBuffer)
 
+let enableFlowControl conn isServer =
+  let q = conn.outputQueue in
+  q.available <- false;
+  Lwt.ignore_result (flushBuffer conn.outputBuffer);
+  q.flowControl <- true;
+  q.canWrite <- isServer;
+  if q.canWrite then
+    Lwt.ignore_result (popOutputQueues q >>= Lwt_unix.yield)
+
```

Comment:
The calls to Lwt.ignore_result look suspicious here, especially the first one. Are you sure you don't want to wait for flushBuffer to be complete before setting q.flowControl to true? Given the contexts in which this function is used, I would turn it into a proper monadic function.

Reply:
I pretty much followed the example of allowWrites and disableFlowControl from the original code. That said, I've now changed it as you proposed (and fixed a bug I found while at it).